src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -104,6 +105,8 @@ public void save(AddressBook addressBook) throws StorageOperationException {
             marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
             marshaller.marshal(toSave, fileWriter);
 
+        } catch (FileNotFoundException fnfe) { 
+        	throw new FileNotFoundException("File path " + path + " not found!");
```

Comment:
This method has the capability create a file is it does not exist. Therefore, the exception will be thrown. So at the beginning of the method check if the file exists and if not throw the exception. 

Then you have to implement a separately a way to handle the exception. The best place to catch and terminate is Main::executeCommand() (this method already has a catch block to terminate upon exception.
Warn but continue: 

If you do not want to continue, without terminating Need to detect the file deletion at the beginning of the StorageFile:: save() method but throw the exception at the end of the method.
It should be caught in Main::executeCommand() but should show a warning to the user instead of terminating the program.